CMakeLists.txt

```diff
@@ -60,10 +60,21 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 
 find_package(YCM REQUIRED)
-find_package(YARP 3.3 REQUIRED)
+find_package(YARP 3.3 REQUIRED COMPONENTS os dev math rosmsg idl_tools)
 find_package(iDynTree 3.1.0 REQUIRED)
 find_package(Eigen3 REQUIRED)
 
+# Optional dependencies
+option(IDYNTREE_YARP_TOOLS_USES_ICUB_MAIN "Compile the tools that depend on icub-main." ON)
+option(IDYNTREE_YARP_TOOLS_USES_QT "Compile the tools that depend on Qt." ON)
+
+if(IDYNTREE_YARP_TOOLS_USES_ICUB_MAIN)
+  find_package(ICUB REQUIRED)
+endif()
+
+if(IDYNTREE_YARP_TOOLS_USES_QT)
+  find_package(Qt5 COMPONENTS Widgets Charts REQUIRED)
+endif()
```

Comment:
Would it be possible to have some basic mechanism to check if those dependencies are actually available before setting the default value of the option? Then, if the user sets it manually, it is totally ok to crash and burn

Reply:
Yes, I wanted to avoid that due to the additonal complexity in CMake, but it is trivial to do so.

Reply:
Done: https://github.com/robotology/idyntree-yarp-tools/pull/12/commits/6d66d4e05ce43e33ef85d644d881ec4c5f661aaf .